common/common.go

```diff
@@ -235,21 +236,45 @@ func GetLightdInfo() (*walletrpc.LightdInfo, error) {
 }
 
 func getBlockFromRPC(height int) (*walletrpc.CompactBlock, error) {
+	// `block.ParseFromSlice` correctly parses blocks containing v5
+	// transactions, but incorrectly computes the IDs of the v5 transactions.
+	// We temporarily paper over this bug by fetching the correct txids via a
+	// verbose getblock RPC call, which returns the txids.
+	//
+	// Unfortunately, this RPC doesn't return the raw hex for the block,
+	// so a second getblock RPC (non-verbose) is needed (below).
+	// https://github.com/zcash/lightwalletd/issues/392
+
 	params := make([]json.RawMessage, 2)
 	heightJSON, err := json.Marshal(strconv.Itoa(height))
 	if err != nil {
 		Log.Fatal("getBlockFromRPC bad height argument", height, err)
 	}
 	params[0] = heightJSON
-	params[1] = json.RawMessage("0") // non-verbose (raw hex)
+	params[1] = json.RawMessage("1") // JSON with list of txids
 	result, rpcErr := RawRequest("getblock", params)
```

Comment:
Can this return a different block from the previous call if there has been a reorg?

Reply:
Well, this _is_ the first call. Protection against a reorg happening between the two calls to `getblock` (for the same block)  is exactly the purpose of this PR. The fix is exactly what you're suggesting here, the first (verbose) call provides the list of transactions and the block hash; the second call fetches the block using block hash so it's guaranteed to be the same.

Your comment points out that the comment is confusing, as it implies that the only reason we're doing this `getblock` call is to get the txids; we're doing it for that reason but also to get the block hash. I just force-pushed to improve the comment.

Reply:
Oops, @defuse, I didn't see your reply before I posted mine.

Reply:
I'm confused as to how I misread this originally. Yes this looks fine.